apinetlet/controllers/network_peering_controller.go

```diff
@@ -0,0 +1,270 @@
+	for _, name := range peeringNames {
+		newStatusPeerings = append(newStatusPeerings, networkingv1alpha1.NetworkPeeringStatus{
+			Name: name,
+		})
+	}
+	network.Status.Peerings = newStatusPeerings
+
+	log.V(1).Info("Updating network status peerings", "", network.Status.Peerings)
+	if err := r.Status().Patch(ctx, network, client.StrategicMergeFrom(base)); err != nil {
+		return fmt.Errorf("error updating network status peerings: %w", err)
+	}
+	log.V(1).Info("Updated network status peerings")
+
+	return nil
+}
+
+func (r *NetworkPeeringReconciler) updateSpec(
+	ctx context.Context,
+	log logr.Logger,
+	network *networkingv1alpha1.Network,
+	peeringClaimRefs []networkingv1alpha1.NetworkPeeringClaimRef,
+	peeredIDs []string,
+) error {
+	if err := r.updateApinetNetworkSpec(ctx, log, network, peeredIDs); err != nil {
+		return fmt.Errorf("error updating apinet network spec: %w", err)
+	}
+
+	base := network.DeepCopy()
+	network.Spec.PeeringClaimRefs = peeringClaimRefs
+	log.V(1).Info("Updating network spec incoming peerings")
+	if err := r.Patch(ctx, network, client.StrategicMergeFrom(base)); err != nil {
+		return fmt.Errorf("error updating network spec incoming peerings: %w", err)
+	}
+
+	log.V(1).Info("Updated network spec incoming peerings")
+	return nil
+}
+
+func (r *NetworkPeeringReconciler) updateApinetNetworkSpec(
+	ctx context.Context,
+	log logr.Logger,
+	network *networkingv1alpha1.Network,
+	peeredIDs []string,
+) error {
+	apinetNetwork := &apinetv1alpha1.Network{}
+	apinetNetworkKey := client.ObjectKey{Namespace: r.APINetNamespace, Name: string(network.UID)}
+	if err := r.APINetClient.Get(ctx, apinetNetworkKey, apinetNetwork); err != nil {
+		return fmt.Errorf("apinet network is not created yet for network: %s \n %w", network.Name, err)
+	}
+
+	base := apinetNetwork.DeepCopy()
+	apinetNetwork.Spec.PeeredIDs = peeredIDs
+	log.V(1).Info("Updating apinet network spec peered ids")
+	if err := r.APINetClient.Patch(ctx, apinetNetwork, client.StrategicMergeFrom(base)); err != nil {
+		return fmt.Errorf("error updating apinet network spec peered ids: %w", err)
+	}
+
+	log.V(1).Info("Updated apinet network spec peered ids")
+	return nil
+}
+
+func (r *NetworkPeeringReconciler) reconcilePeering(
+	ctx context.Context,
+	log logr.Logger,
+	network *networkingv1alpha1.Network,
+	peering networkingv1alpha1.NetworkPeering,
+) (networkingv1alpha1.NetworkPeeringClaimRef, string, string, error) {
+	networkKey := client.ObjectKeyFromObject(network)
+
+	targetNetwork := &networkingv1alpha1.Network{}
+	targetNetworkRef := peering.NetworkRef
+	targetNetworkNamespace := targetNetworkRef.Namespace
+	if targetNetworkNamespace == "" {
+		targetNetworkNamespace = network.Namespace
+	}
+	targetNetworkKey := client.ObjectKey{Namespace: targetNetworkNamespace, Name: targetNetworkRef.Name}
+	log = log.WithValues("TargetNetworkKey", targetNetworkKey)
+
+	log.V(2).Info("Getting target network")
```

Comment:
is there a reason why this is loglevel=2? I would consider this as loglevel=1

Reply:
Done